script/translate.go

```diff
@@ -102,12 +104,51 @@ func inTranslate(n *html.Node, filename string) {
 	}
 }
 
-func translation(v string) {
+func isNamespacedTranslationId(id string) bool {
+	if !strings.Contains(id, ".") {
+		return false
+	}
+
+	if id[len(id)-1] == '.' {
+		// Last section of namespaced ID must not be empty
+		return false
+	}
+
+	if strings.Contains(id, " ") {
+		// Probably a sentence
+		return false
+	}
+
+	if strings.Contains(id, "..") {
+		// Probably something like "Loading..."
+		return false
+	}
+
+	return true
```

Comment:
This is really a heuristic with a lot of guesswork and potential for breakage.  If we want to employ structured IDs, there would need to be some clear rules on the semantics, without conflicts regarding the previously used keys.

Reply:
I decided this heuristic isn't needed, as the namespace parsing should only apply to translations with an explicitly specified translation key, so I deleted it. But either way, the translation enhancements are moved to PR #9192.